File: app/src/main/java/com/rock/moduleinit/AppApplication.java
package com.rock.moduleinit;

import android.app.Application;
import android.content.Context;

import org.jetbrains.annotations.Nullable;

public class AppApplication extends Application {

    @Override
    protected void attachBaseContext(@Nullable Context base) {
        super.attachBaseContext(base);
//        DemoDelegateApplication.getInstance().attachBaseContextMethodExit((Application) this);
    }

    @Override
    public void onCreate() {
//        DemoDelegateApplication.getInstance().onCreateMethodEnter();
        super.onCreate();
//        DemoDelegateApplication.getInstance().onCreateMethodExit();
    }
}

File: app/src/main/java/com/rock/moduleinit/AppApplication.kt
package com.rock.moduleinit

import android.app.Application
import android.content.Context
import android.util.Log
import com.rock.democomponent.DemoDelegateApplication

class AppApplication(private var mDemoDelegateApplication: DemoDelegateApplication = DemoDelegateApplication()) :
    Application() {

    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)
    }

    override fun onCreate() {
        mDemoDelegateApplication.onCreateMethodEnter()
        super.onCreate()
        Log.d("MuXi", "onCreate" + " : : ");
        mDemoDelegateApplication.onCreateMethodExit()
    }
}